VultisigApp/VultisigApp/Services/CoinService/EvmCoinService.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  EvmCoinService.swift
+//  VultisigApp
+//
+//  Created by Enrique Souza Soares on 05/07/24.
+//
+
+import Foundation
+
+@MainActor
+class EvmCoinService: CoinService {
+    
+    override func addDiscoveredTokens(nativeToken: Coin, to vault: Vault) async {
+        do {
+            let service = try EvmServiceFactory.getService(forCoin: nativeToken)
+            let tokens = await service.getTokens(nativeToken: nativeToken)
+            
+            for token in tokens {
+                do {
+                    _ = try await addToChain(asset: token, to: vault, priceProviderId: nil)
+                } catch {
+                    print("Error adding the token \(token.ticker) service: \(error.localizedDescription)")
+                }
+            }
+        } catch {
+            print("Error fetching service: \(error.localizedDescription)")
+        }
+    }
+    
+    override func addToChain(assets: [CoinMeta], to vault: Vault) async throws {
```

Comment:
Honestly I am not a fan of this , see `addToChain` , this method exist on many places....  two in it's parent.  And override here as well, it is super confusing ,  you need to think about it multiple times to figure which method it is actually calling...

Reply:
What is your suggestion?

Reply:
maybe don't put any implementation on the parent class `CoinService`

Reply:
What about the code duplication?